### ISSUE TITLE
Download themes without logging in

### ISSUE THREAD

**Issue by malitech** [Nordskog/SwiftKeyExi#48]
Is there a way to download themes without logging in? I don't want to give internet permission to swiftkey.

**Comment by Nordskog** [comment]
Download themes without logging would be actually be very straight forward, although it's not something I am going to implement or otherwise condone. Like it or not, it's their business model at this point. 

You would still need an internet connection actually download them though, even if you circumvent the login requirement.

You might have some luck manually copying themes ( they are saves in one of a few theme folders in the app's files directory ), though just copying them directly doesn't seem to work.

**Comment by timikesmarky** [comment]
> Download themes without logging would be actually be very straight forward, although it's not something I am going to implement or otherwise condone. Like it or not, it's their business model at this point.
> 
> You would still need an internet connection actually download them though, even if you circumvent the login requirement.
> 
> You might have some luck manually copying themes ( they are saves in one of a few theme folders in the app's files directory ), though just copying them directly doesn't seem to work.

Thanks for the time. I have never logged in or given internet or account permissions to swiftkey with appops and system permissions. It's also blocked with firewall.

Anyway, I downloaded all themes from swiftkey in emulator and I copied the theme files to my device and I was playing with it in the app directory and force stopping it and it wasn't working. Suddenly, I saw that I have ALL the themes in Swiftkey. I don't know how it happened. I immediately created a titanium backup. Now I have all themes without logging in or giving any permissions. 💃 I was sooo thrilled and happy. I backed up the backup too so that I can use it on any phone.

**Comment by Nordskog** [comment]
Oooh, so that means it... is possible!
I tried copying over the individual folders and themes, and editing the theme list json files to match, but alas had little luck. 

Not that I know that it is possible I might put in a bit more effort to try and get some concrete steps recorded.

Thanks for letter me know, and sorry about the delay in replying.

**Comment by timikesmarky** [comment]
Yeah. I haven't had time to check and find out how it happened. I also tried editing the json files at first. Nothing was working. I don't know what the heck happened. I suddenly got all the themes. My mouth was literally open.

Thank you. Please let me know if you find something. :)
